### PR TITLE
Fix test race, and remove JujuConnSuite dependency.

### PR DESCRIPTION
**Pull request by howbazaar**
This branch fixes a test race that was introduced in the recent PR #7709.

As a drive by I removed the JujuConnSuite and introduced an interface for the API calls.

### CODE REVIEW

File: worker/logger/logger_test.go [hunk 1]
```diff
@@ -78,21 +79,19 @@ func (s *LoggerSuite) TestRunStop(c *gc.C) {
...
 	loggo.DefaultContext().ResetLoggerLevels()
-	err = loggo.ConfigureLoggers(initial)
+	err := loggo.ConfigureLoggers(initial)
 	c.Assert(err, jc.ErrorIsNil)
 
 	loggingWorker := s.makeLogger(c)
-	defer worker.Stop(loggingWorker)
-
 	s.waitLoggingInfo(c, expected)
+	worker.Stop(loggingWorker)
```
Comment (wallyworld) [inline worker/logger/logger_test.go]: let's check that err is nil here
why remove the defer?

File: worker/logger/logger_test.go [hunk 2]
```diff
@@ -110,3 +109,35 @@ func (s *LoggerSuite) TestConfigOverride(c *gc.C) {
...
+type mockAPI struct {
+	watcher *mockNotifyWatcher
+	config  string
+}
+
+func (m *mockAPI) LoggingConfig(agentTag names.Tag) (string, error) {
+	return m.config, nil
+}
+func (m *mockAPI) WatchLoggingConfig(agentTag names.Tag) (watcher.NotifyWatcher, error) {
+	return m.watcher, nil
```
Comment (wallyworld) [inline worker/logger/logger_test.go]: we should set up the mock api with the expected agent and then assert here that the agent matches to make sure we are passing in the expected value, or just assert that it equals 42 as that's what was set up earlier

File: worker/logger/logger_test.go (hunk 1 above)
Comment (howbazaar) [inline worker/logger/logger_test.go]: That was the race. The worker may update the value in another goroutine. Stop the worker before confirming the expected callback.

File: worker/logger/logger_test.go (hunk 2 above)
Comment (howbazaar) [inline worker/logger/logger_test.go]: We can record the agent tag passed in and confirm in the test, but the gc.C that we have in the setup doesn't work properly within the normal tests.

**Comment by howbazaar** [comment]
$$merge$$

**Pull request merged**